internal/ingress/controller/kong.go

```diff
@@ -72,8 +74,28 @@ func (n *KongController) OnUpdate(ctx context.Context, state *kongstate.KongStat
 	} else {
 		err = n.onUpdateDBMode(targetContent)
 	}
+	var target []byte
+	if n.cfg.DumpConfig == util.ConfigDumpModeSensitive {
+		target, _ = json.Marshal(targetContent)
+	} else if n.cfg.DumpConfig == util.ConfigDumpModeEnabled {
+		sanitizedState := state.SanitizedCopy()
+		target, _ = json.Marshal(n.toDeckContent(ctx, sanitizedState))
+	}
 	if err != nil {
+		if len(target) > 0 {
+			dumpErr := ioutil.WriteFile(filepath.Join(n.dumpDir, "target.json"), target, 0600)
+			if dumpErr != nil {
+				n.Logger.Warnf("failed to dump configuration: %s", dumpErr)
+			}
+		}
 		return err
+	} else {
+		if len(target) > 0 {
+			dumpErr := ioutil.WriteFile(filepath.Join(n.dumpDir, "last_good.json"), target, 0600)
+			if dumpErr != nil {
+				n.Logger.Warnf("failed to dump configuration: %s", dumpErr)
+			}
```

Comment:
These two branches (`err != nil` and `else`) differ only by a string literal. There's no need to duplicate the logic because of that. How about creating a helper function:
```go
func buildDumpPath(runFailed bool, dumpDir string) {
	if runFailed {
		return filepath.Join(dumpDir, "target.json") // I don't understand the thought behind the name "target.json" but let's not bikeshed
	}
	return filepath.Join(dumpDir, "last_good.json")
}
```

and replacing both branches with this `if` with something streamlined, for example:
```go
dumpPath := buildDumpPath((err != nil), n.dumpDir)
if dumpErr := ioutil.WriteFile(dumpPath, target, 0600); dumpErr != nil {
	n.Logger.WithField("dump_path", dumpPath).WithError(dumpErr).Warnf("failed to dump configuration)
}
```
?